include/circt/Dialect/StaticLogic/StaticLogic.td

```diff
@@ -141,8 +141,20 @@ def PipelineWhileOp : Op<StaticLogic_Dialect, "pipeline.while", []> {
   ];
 
   let extraClassDeclaration = [{
+    // Get the condition block.
     Block &getCondBlock() { return condition().front(); }
+
+    // Get the stages block.
     Block &getStagesBlock() { return stages().front(); }
+
+    // Get the terminator of the pipeline.
+    PipelineTerminatorOp getTerminator();
+
+    // Compute the distance between the two stages.
+    uint64_t computeDistance(PipelineStageOp, PipelineStageOp);
```

Comment:
Hmm, I think "distance" is intuitively understandable here, and I don't have a better suggestion, but the term does have a different meaning in the context of the cyclic scheduling problem.

Reply:
I will think on it and try to use a better name.

Reply:
went with `computeNumStages`, since this "computes the number of stages" between the two.